### DISCUSSION TITLE
v20.0.0-beta.2

### DISCUSSION THREAD

**Discussion by jonyo** [geodesicsolutions-community/geocore-community#42] [Announcements]
### Since beta 1

* Fix several PHP deprecation notices in the admin panel. by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/30
* Fixing build to include user_images, also seperating extra template sets by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/35
* Update version number to 20.0.0 in software by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/38
* Add a link to the license agreement on github on install and upgrade pages by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/39
* Update old copyright notices to say distributed freely under MIT license by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/40

### Since last legacy code release 18.02.0

* Import legacy code - this imports the code with a fresh start from latest release by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/7
* Remove log license addon by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/8
* Apply all the un-released changes from legacy codeset by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/10
* Remove OS files and add them to .gitignore by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/11
* Update installation requirements check and links by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/14
* Add docker support by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/17
* Update the admin panel wiki links to go to the new URL by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/19
* Fleshing out the docker instructions by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/20
* Add docker install link by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/21
* Fix the error on the cron admin page by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/23
* Getting ready to make a release package by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/25
* Fix several PHP deprecation notices in the admin panel. by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/30
* Fixing build to include user_images, also seperating extra template sets by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/35
* Update version number to 20.0.0 in software by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/38
* Add a link to the license agreement on github on install and upgrade pages by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/39
* Update old copyright notices to say distributed freely under MIT license by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/40

## New Contributors
* @jonyo made their first contribution in https://github.com/geodesicsolutions-community/geocore-community/pull/7

**Full Changelog**: https://github.com/geodesicsolutions-community/geocore-community/compare/v0.0.0...v20.0.0-beta.2

<hr /><em>This discussion was created from the release <a href='https://github.com/geodesicsolutions-community/geocore-community/releases/tag/v20.0.0-beta.2'>v20.0.0-beta.2</a>.</em>

**Comment by jonyo** [comment]
If you notice, I went ahead and switched over to the suggested v20.

Also I made it separate out the extra template sets.  You'll see them in the release assets as individual downloads.  That was needed because we don't warn against changing those, and if it were part of the main package it would overwrite anything they customized in those.

**Reply by vicos59** [reply]
I think v20 will work well. It provides continuity, 20 is easy to remember that this was the beginning of CE, and it provides an air gap between the last commercial version.

I will upgrade my Beta-1 install and let you know how it goes.

**Reply by vicos59** [reply]
I'm dead in the water with the upgrade from Beta-1. Something wrong with the PHP requirement check.

![bug-3](https://user-images.githubusercontent.com/6634041/152710333-7348b558-fda0-47dd-963f-32b411aba062.JPG)

**Reply by jonyo** [reply]
Well the "displayed requirement" is out-dated, it is actually checking that you have at least PHP 7.4 and it shows you are on PHP 7.2.34 - are you able to run PHP 7.4?

I'll fix what it shows.  Basically when the version check fails, it shows a dummy requirement check page so that other things the upgrade script does, does not break the page from showing at all.  Anyways, I missed updating what the dummy "you don't have high enough version of PHP" template says for min version and all that.

It is not "breaking" since it works (at least for me) if you do have PHP 7.4, so I won't rush to update the display, but I'll make sure it gets fixed for net next beta.

**Comment by vicos59** [comment]
As an aside, which of the template sets provided best adheres to modern standards, i.e. responsive web design and mobile friendly?

**Reply by jonyo** [reply]
I do not know about the extras, but pretty sure the main one does.  The others should too but I would not be surprised if they did not, I did not have a hand in making them.  But at the time they were made, responsive design / mobile friendly was all the rage.  (Now it is just kinda normal, back then responsive design was a new concept).

tl;dr: probably all of them are fine.

**Comment by vicos59** [comment]
The redirect to SSL is not working.

If I go to http://site.com it should redirect to https://site.com

Instead the browser return an ERR_INVALID_REDIRECT

The http header output is:

`
< HTTP/1.1 302 Found
< Date: Mon, 07 Feb 2022 06:06:43 GMT
< Server: Apache
< Cache-Control: no-cache, must-revalidate
< Expires: Sat, 26 Jul 1997 05:00:00 GMT
< Set-Cookie: classified_session=09e805489f31f7e729b48cca7d2f18ab; path=/; domain=.[snip].com

< Location: https:/

< Vary: User-Agent
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
`

The Location header directive is not correct. Pretty sure this worked in Beta1...could be wrong.

**Reply by vicos59** [reply]
OK, it looks like this is caused when you do not add 'index.php' to the 'Site URL' and 'SSL Site URL' in General Settings.

**Reply by jonyo** [reply]
Yeah that is just how those settings work.

This may be a good thing to overhaul, this was done back when SSL was not as common.  These days though, you can get an SSL free with automated tools.

So it may be time to update how this part works, to not try to do "per page" switching back and forth between SSL or not... But a site wide thing where it is either on or off.  And if it is on, and you go to http it would automatically take you to https.

**Comment by vicos59** [comment]
When a user loads a page, GeoCore tries to run **cron.php?action=cron** in the background. This fails:

`
Input array has 1 params, does not match query: 'DELETE FROM `geodesic_addon_storefront_subscriptions` WHERE `expiration` < 1643955181 AND `onhold_start_time` = 0'
`

You can see this using the browser Inspector in the Console tab (Chromium based browsers).

Disabled the Storefront addon and then cron.php fails with a HTTP 500 error.

**Reply by jonyo** [reply]
Made #49 thanks.

**Reply by jonyo** [reply]
So when I was looking into this, I saw, there was no way it worked as-is.  So I was thinking it must be in some of the brand new code in new "features never released" that were added after 18.2.0.

Turns out, nope, the line in question is an old one.  In fact it has not changed since that file was created, 13 years ago 😆 - @WOs8lfTkz3E5g7I4Lm  you are finding 13 year old bugs that no one has caught yet!  🥳 

It'll be fixed in next beta.

**Reply by vicos59** [reply]
Cool, let me know when the next Beta is out and I will update my Beta-2. When you feel you are more close to a stable release, I will try an update of my 7.5.1 test site to see if that goes well.

**Reply by vicos59** [reply]
I am also going to try and test the Facebook connect today. I know they have done quite a few mandatory API updates in recent years.

**Reply by vicos59** [reply]
Now that I think about it, my mariaDB was just upgraded and that enabled STRICT mode. So, that might be why it started throwing errors. In any case, good to fix it.

**Reply by jonyo** [reply]
FYI speaking of mariaDB - that is what I'm testing on myself (using docker container).

> Cool, let me know when the next Beta is out

I was holding off a bit since the things discovered so far are "non breaking" (they are all minor or have easy fixes).  Since it's been a few days though I can generate a new release, maybe during my lunch break today if I have time (CST timezone here).  I was also getting some "house cleaning" changes in as I have time.  

With the "house cleaning" (aka #6 ) I have a lofty goal: to be able to run the automated PHP code "sniffer" (called `phpcs` if you are familiar) using PSR12 rules, and it come back with 0 errors.  Once it is to that point, we can turn it on as a requirement to run before it will let you merge a PR.  That will go a long way toward making the code much more stable, as it reduces / removes the "silly mistakes" (or sometimes not so silly) that human eyes can miss easily.

The trouble is the age of the code, back then code sniffers were not so common, so there is a lot of things to "clean up".  The sniffer itself can actually do a ton of it automatically for you, and I already had it do that stuff, but it can't make a line shorter (for instance) on its own.  So I'm going through the reported errors and fixing them in chunks at a time.  But I'm not wanting to do it all before the next beta, just saying that is something I'm working on in between.

**Reply by vicos59** [reply]
Understood. I have spent half of my life cleaning up stuff.

Lat time I did any serious code, I was using SVN and Komodo. I just installed Docker. What other current tools do you use/suggest? Once I get up to speed on the modern environment, I can chip in more.

**Reply by jonyo** [reply]
For development, the IDE I'd suggest these days is Visual Studio Code (VSC).  If you go that route, once you have it these are my extension recommendations for working with PHP:

* `EditorConfig for VSCode` - for following the `.editorconfig` config, see https://editorconfig.org/ for info
* `GIT Lense - GIT Supercharged`
* `PHP Debug` - for xdebug - the one by `Xdebug` publisher
* `PHP Intelephense` (sic) - that is the correct spelling.  Note that there is a paid version but you can use it for free just with some of the more advanced features disabled.  It works much better than the `PHP Intelesense` made by someone else.

Also, I like to use `GIthub desktop` for viewing changes only - I don't do actual commits with it because in some cases, it does some "magic" for you and does more than you tell it to do.  I prefer to make actual changes on the command line where it only does what you tell it to, no auto merging and things like that unless you tell it to.

Umm, what else...  If you are on mac, `Boop` is handy 1-off app for doing string manipulation... Like if you have a long JSON text you can paste it in Boop and tell it to format JSON and it will format it for you.  Stuff like that, it comes in handy sometimes as an alternative to letting some random site do it for you which could be problematic if the text contains sensitive info like passwords.

For DB stuff I tend to use TablePlus (again if you are on a mac, maybe they have windows version I dunno).

Spotify to keep me motivated 😂

**Reply by jonyo** [reply]
One thing I've been thinking of doing is make a video for how to set up docker and get it working from scratch, from "nothing installed" all the way to having GeoCore running on localhost:8080.  I have a feeling folks shy away from it because it seems complicated.  But once you have it installed and working it makes things really easy for testing on your local computer, no need to set up WAMP or use a test host.

**Reply by vicos59** [reply]
A quick video would be helpful..
